server/command/command.go

```diff
@@ -0,0 +1,285 @@
+package command
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/bakurits/mattermost-plugin-splunk/server/api"
+	"github.com/bakurits/mattermost-plugin-splunk/server/config"
+	"github.com/bakurits/mattermost-plugin-splunk/server/splunk"
+
+	"github.com/google/uuid"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+const (
+	helpTextHeader = "###### Mattermost Splunk Plugin - Slash command help\n"
+	helpText       = `
+* |/splunk help| - print this help message
+* |/splunk auth --login [server base url] [username] [password]| - log into the splunk server
+* |/splunk alert --subscribe| - subscribe to alerts
+* |/splunk logs --list| - list names of logs on server
+* |/splunk log [logname]| - show specific log from server
+`
+	autoCompleteDescription = ""
+	autoCompleteHint        = ""
+	pluginDescription       = ""
+	slashCommandName        = "splunk"
+)
+
+// Handler returns API for interacting with plugin commands
+type Handler interface {
+	Handle(args ...string) (*model.CommandResponse, error)
+}
+
+// HandlerFunc command handler function type
+type HandlerFunc func(args ...string) (*model.CommandResponse, error)
+
+// HandlerMap map of command handler functions
+type HandlerMap struct {
+	handlers       map[string]HandlerFunc
+	defaultHandler HandlerFunc
+}
+
+// NewHandler returns new Handler with given dependencies
+func NewHandler(args *model.CommandArgs, conf *config.Config, a splunk.Splunk) Handler {
+	return newCommand(args, conf, a)
+}
+
+// GetSlashCommand returns command to register
+func GetSlashCommand() *model.Command {
+	return &model.Command{
+		Trigger:          slashCommandName,
+		DisplayName:      slashCommandName,
+		Description:      pluginDescription,
+		AutoComplete:     true,
+		AutoCompleteDesc: autoCompleteDescription,
+		AutoCompleteHint: autoCompleteHint,
+	}
+}
+
+func (c *command) Handle(args ...string) (*model.CommandResponse, error) {
+	ch := c.handler
+	if len(args) == 0 || args[0] != "/"+slashCommandName {
+		return ch.defaultHandler(args...)
+	}
+	args = args[1:]
+
+	for n := len(args); n > 0; n-- {
+		h := ch.handlers[strings.Join(args[:n], "/")]
+		if h != nil {
+			return h(args[n:]...)
+		}
+	}
+	return ch.defaultHandler(args...)
+}
+
+// command stores command specific information
+type command struct {
+	args    *model.CommandArgs
+	config  *config.Config
+	splunk  splunk.Splunk
+	handler HandlerMap
+}
+
+func (c *command) help(_ ...string) (*model.CommandResponse, error) {
+	helpText := helpTextHeader + helpText
+	return c.postCommandResponse(helpText), nil
+}
+
+func (c *command) postCommandResponse(text string) *model.CommandResponse {
+	return &model.CommandResponse{Text: text}
```

Comment:
If this function is implemented this way, the ephemeral post will be made by the plugin's bot, and not `System`.

```suggestion
post := &model.Post{
	ChannelId: c.args.ChannelId,
	UserId: c.splunk.BotUser()
	Message:   text,
}
_ = c.splunk.SendEphemeralPost(c.args.UserId, post)
return &model.CommandResponse{}
```

Reply:
Made an issue https://github.com/mattermost/mattermost-plugin-splunk/issues/30